src/rpm.rs

```diff
@@ -1253,7 +1387,122 @@ impl RPMBuilder {
         self
     }
 
-    pub fn build(mut self) -> Result<RPMPackage, RPMError> {
+    /// build without a signature
+    ///
+    /// ignores a present key, if any
+    pub fn build(self) -> Result<RPMPackage, RPMError> {
+        let (lead, header_idx_tag, content) = self.prepare_data()?;
+
+        let mut header = Vec::with_capacity(128);
+        header_idx_tag.write(&mut header)?;
+        let header = header;
+
+        let (
+            header_digest_sha1,
+            header_and_content_digest_md5,
+        ) = Self::derive_hashes(header.as_slice(), content.as_slice())?;
+
+        let header_and_content_len = header.len() + content.len();
+
+        let digest_header = Header::<IndexSignatureTag>::builder()
+            .add_digest(
+                header_digest_sha1.as_str(),
+                header_and_content_digest_md5.as_slice(),
+            )
+            .build(header_and_content_len as i32);
+
+        let metadata = RPMPackageMetadata {
+            lead,
+            signature: digest_header,
+            header: header_idx_tag,
+        };
+        let pkg = RPMPackage { metadata, content };
+        Ok(pkg)
+    }
+
+    /// use an external signer to sing and build
+    ///
+    /// See `crypto::Signing` for more details.
+    #[cfg(feature = "signing-meta")]
+    pub fn build_and_sign<S>(self, signer: S) -> Result<RPMPackage, RPMError>
+    where
+        S: crypto::Signing<crate::crypto::algorithm::RSA>,
+    {
+        let (lead, header_idx_tag, content) = self.prepare_data()?;
+
+        let mut header = Vec::with_capacity(128);
+        header_idx_tag.write(&mut header)?;
+        let header = header;
+
+        let (
+            header_digest_sha1,
+            header_and_content_digest_md5,
+        ) = Self::derive_hashes(header.as_slice(), content.as_slice())?;
+
+        let header_and_content_len = header.len() + content.len();
+
+        let builder = Header::<IndexSignatureTag>::builder().add_digest(
+            header_digest_sha1.as_str(),
+            header_and_content_digest_md5.as_slice(),
+        );
+
+        let signature_header = {
+            let rsa_sig_header_only = signer.sign(header.as_slice()).map_err(|_e| {
+                dbg!(_e);
```

Comment:
I think we should remove the dbg! statements to make this less noisy for library users during development

Reply:
Whoops